scipy/stats/stats.py

```diff
@@ -2306,6 +2311,186 @@ def zmap(scores, compare, axis=0, ddof=0):
         return (scores - mns) / sstd
 
 
+def _iqr_scale_to_num(scale):
+    """
+    A helper function to convert the 'scale' argument of `iqr` to a number.
+
+    scale: str
+        If scale is one of ['raw', 'normal'], this function will map the string
+        to a value.
+
+    The concept for this function is borrowed heavily from the `numpy.histogram`
+    helper function `_hist_optim_numbins_estimator`.
+    """
+    assert isinstance(scale, string_types)
```

Comment:
No ``assert`` please, because this line may disappear after byte-compiling with ``-O``. You want

    if not isinstance(scale, string_types):
        raise informative_error